libmscore/undo.cpp

```diff
@@ -318,6 +331,7 @@ void UndoStack::rollback()
 
 void UndoStack::endMacro(bool rollback)
       {
+      mutexPushPop.lock();
       if (curCmd == 0) {
             qWarning("not active");
             return;
```

Comment:
unlock missing here

Reply:
or just lock only after this early exit

Reply:
you are right...I missed that.  Actually Joshua just informed me of a useful convenience class QMutexLocker (https://doc.qt.io/qt-5/qmutexlocker.html) which does exactly what I need here and avoids such mistakes.

Reply:
Ah, that'd make it far easier indeed.
Should probably get used in some more places thoughout the MuseScore code